iron-fit-behavior.html

```diff
@@ -600,23 +639,24 @@
           continue;
         }
 
-        position = position || pos;
         pos.croppedArea = this.__getCroppedArea(pos, size, fitRect);
-        var diff = pos.croppedArea - position.croppedArea;
-        // Check which crops less. If it crops equally, check if align is ok.
-        if (diff < 0 || (diff === 0 && alignOk)) {
-          position = pos;
-        }
         // If not cropped and respects the align requirements, keep it.
         // This allows to prefer positions overlapping horizontally over the
         // ones overlapping vertically.
-        if (position.croppedArea === 0 && alignOk) {
+        if (pos.croppedArea === 0 && alignOk) {
+          position = pos;
```

Comment:
I have to say `position` and `pos` are pretty bad names. Maybe something along the lines of `finalPosition` || `outPosition` || `acualPosition` and `currentPosition

Reply:
Also, doesn't this section break API? position will always return as pos instead of `position = position || pos` which was moved below this if statement

Reply:
Yeah, I agree that the naming is not great :x Will update.

Regarding the change: `alignOk` is computed from `pos`. Previously, we were looking `position.croppedArea` instead of `pos.croppedArea`, which was basically like saying `positions[1] has the correct align properties and positions[0] doesn't crop. Let's pick positions[0]!`.

In this change I wanted to make it clear that if the current position doesn't crop and has the correct alignments, we use that. Otherwise, we use search for the best position that doesn't crop (see lines below...)